src/lib/components/menu/wrappers/WithSelectMenu.svelte

```diff
@@ -0,0 +1,144 @@
+      // single-select: do nothing if already selected
+      if (!multiple && isSelected(selections, key)) {
+        return;
+      }
+      // set temporarily selected to get the icon to change instantly, then wait for tick
+      // proceed with rest of update
+      if (multiple) {
+        // check to see if exists
+        // if not, add.
+        if (isSelected(selections, key)) {
+          selections = [...selections.filter((s) => s.key !== key)];
+        } else {
+          selections = [
+            ...selections,
+            { main, right, key, description, index },
+          ];
+        }
+      } else {
+        // replace selected with a single value.
+        selections = [{ main, right, description, key, index }];
+      }
+      dispatch("select", selections);
+      if (!multiple) closeEventHandler();
+
+      temporarilySelectedKey = undefined;
+    };
+  }
+
+  function isSelected(selections, key) {
+    return selections?.some((selection) => selection.key === key);
+  }
+</script>
+
+<WithTogglableFloatingElement
+  bind:active
+  {location}
+  {alignment}
+  {distance}
+  let:handleClose
+  let:toggleFloatingElement
+>
+  <slot {active} {handleClose} toggleMenu={toggleFloatingElement} />
+
+  <Menu
+    slot="floating-element"
+    {dark}
+    on:lose-focus={() => {
+      if (active) handleClose();
+    }}
+    on:escape={handleClose}
+  >
+    {#each options as { key, main, description, right }, i}
+      {@const selected = isSelected(selections, key)}
+      <MenuItem
+        icon
+        animateSelect={!multiple}
+        on:before-select={() => {
+          temporarilySelectedKey = key;
+        }}
+        on:select={createOnClickHandler(
+          main,
+          right,
+          description,
+          key,
+          i,
+          handleClose
+        )}
+        {selected}
+      >
+        <svelte:fragment slot="icon">
+          <!-- this conditional will make the circle check appear briefly before the menu closes
+          in the case of a single-select menu. -->
+          {#if !multiple}
+            {#if (temporarilySelectedKey !== undefined && temporarilySelectedKey === key) || (temporarilySelectedKey === undefined && selected)}
+              <!-- {#if style === "obvious"}<CheckCircle />{:else}<Check />{/if} -->
+              <Check />
+            {:else}
+              <Spacer />
+              <!-- {:else if style === "obvious"}
+              <EmptyCircle />{:else}<Spacer /> -->
+            {/if}
+          {:else if selected}
+            {#if style === "obvious"}<CheckBox />{:else}<Check />{/if}
+          {:else if style === "obvious"}<EmptyBox />{:else}<Spacer />{/if}
+        </svelte:fragment>
+
+        {main}
+        <svelte:fragment slot="description">
+          {#if description}
+            {description}
+          {/if}
+        </svelte:fragment>
+        <svelte:fragment slot="right">
+          {right || ""}
+        </svelte:fragment>
+      </MenuItem>
```

Comment:
I am also thinking how we are going to handle multi level inputs/selects down the line. For example if you want to select Currency as the format type, we should have a convenient way of also taking the currency symbol (such as $, €) as a secondary input. Could be multi level menus or just a text box on the right side. Probably we can directly use the `MenuItem` and `Menu` component to create such custom structures. This doesn't have to be necessarily addressed in this PR, just adding for later reference.

Reply:
Yeah, I would like to address multi-level menus at some point. They're technically not hard to do, but I want to be mindful of doing it right (e.g. the "acceptable triangle" approach to keeping them open w/ mousemove)